src/wallet/wallet.cpp

```diff
@@ -4448,10 +4448,18 @@ void CMerkleTx::SetMerkleBranch(const uint256& block_hash, int posInBlock, int b
 
 int CMerkleTx::GetDepthInMainChain(interfaces::Chain::Lock& locked_chain) const
 {
+    if (pwallet == nullptr)
+        return 0;
+
     if (hashUnset())
         return 0;
```

Comment:
In commit "Use CWallet::m_last_block_processed_height in GetDepthInMainChain" (233ae92f2d51333f158810d85d13cf63a1144e6b)

Can we `assert(m_block_height > 0)` after this point? I'm afraid of another bug like the `importprunedfunds` one mentioned earlier leaving m_block_height unset and this function returning bogus values.

Reply:
Hmmm, I introduced  ```assert(m_block_height >= -1)``` as we may GetDepthInMainChain on fresh transactions, their ```hashBlock``` being unset, their depth should appear as zero.

Reply:
Note : I've seen one Travis failure on feature_reindex.py where the assert failed. I wasn't able to reproduce the failure on my own env. I think that's caused by the slow Travis setup but that's not great, what's your take on it ?